### PR TITLE
[KYUUBI #937] Push Kyuubi docker image

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

Push docker image if we commit to master

### _How was this patch tested?_
test in forked repo.

### CODE REVIEW

**Comment by timothy65535** [comment]
Nice Idea.

File: .github/workflows/image.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+jobs:
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
```
Comment (timothy65535) [inline .github/workflows/image.yml:20]: `secrets.DOCKERHUB_TOKEN` to `secrets.DOCKERHUB_PASSWORD`

https://docs.github.com/en/actions/reference/encrypted-secrets

File: .github/workflows/image.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: Publish Docker image
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
Comment (kezhenxu94) [inline .github/workflows/image.yml:7]: Pushing docker images for every pull request is not a good practice, it’s better to push the images only when the commit is pushed to master branch.

Comment (yaooqinn) [reply .github/workflows/image.yml:7]: +1

Comment (ulysses-you) [reply .github/workflows/image.yml:7]: agree, the workflow is only for test now

**Comment by ulysses-you** [comment]
The workflow it now works in my forked repo and docker hub. We can see how the github action works after merged this PR.

**Comment by ulysses-you** [comment]
let me it merge to master first and see what will happen

**Comment by kezhenxu94** [comment]
> let me merge it to master first and see what will happen

You don't need to merge it into master just for testing. You can create a new branch on this repo (not your fork repo) and make the workflow run in that repo.

Also, the docker repo must be the same as the GitHub repo name `apache/incubator-kyuubi`, not `apache/kyuubi`

**Comment by ulysses-you** [comment]
thank you @kezhenxu94 , seems the GA failed with 
```
2021-09-23T02:15:35.6185062Z error: denied: requested access to the resource is denied
2021-09-23T02:15:36.2512609Z ##[error]buildx failed with: error: denied: requested access to the resource is denied
```

Not sure why it happened, I will revert the commit and use new branch to test

**Pull request closed without merging**